corehq/apps/users/views/__init__.py

```diff
@@ -334,6 +340,36 @@ def update_user(self):
         if self.form_user_update.is_valid():
             return self.form_user_update.update_user()
 
+    @property
+    @memoized
+    def tableau_form(self):
+        try:
+            if self.request.method == "POST" and self.request.POST['form_type'] == "tableau":
+                return TableauUserForm(self.request.POST,
+                                    request=self.request,
+                                    domain=self.domain,
+                                    username=self.editable_user.username)
+
+            tableau_user = TableauUser.objects.filter(server__domain=self.domain).get(
+                username=self.editable_user.username
+            )
+            return TableauUserForm(
+                domain=self.domain,
+                request=self.request,
+                username=self.editable_user.username,
+                initial={
+                    'role': tableau_user.role
+                }
+            )
+        except (TableauAPIError, TableauUser.DoesNotExist) as e:
+            messages.error(self.request, _('''There was an error getting data for this user's associated Tableau
+                                             user. Please contact support if this error persists.'''))
```

Comment:
won't there be a situation initially when the tableau user doesn't exist?

Reply:
I mean for existing web users that don't have corresponding TableauUser objects (assuming that's possible)